src/main/java/be/seeseemelk/mockbukkit/plugin/PluginManagerMock.java

```diff
@@ -544,15 +547,158 @@ public boolean isPluginEnabled(Plugin plugin)
+
+					//acknowledge plugin name restrictions
+					String pluginName = pdf.getName();
+					checkPluginNameValidity(file, pluginName);
+
+					//check if already registered
+					File previousRegisteredPlugin = pluginsToLoad.put(pluginName, file);
+					if (previousRegisteredPlugin != null) 
+					{
+						throw new InvalidDescriptionException(String.format("Plugin %s wasn't loaded: Duplicate plugin name %s!", file.getName(), pluginName));
+					}
+
+					//gather softDependencies, dependencies, and loadBefore
+					Set<String> pluginSoftDependencies = softDependencies.getOrDefault(pluginName, new HashSet<>());
+					pluginSoftDependencies.addAll(pdf.getSoftDepend());
+					softDependencies.put(pluginName, pluginSoftDependencies);
+
+					Set<String> pluginDependencies = dependencies.getOrDefault(pluginName, new HashSet<>());
+					pluginDependencies.addAll(pdf.getDepend());
+					dependencies.put(pluginName, pluginDependencies);
+
+					for (String loadBefore : pdf.getLoadBefore())
+					{
+						Set<String> softDepends = softDependencies.getOrDefault(loadBefore, new HashSet<>());
+						softDepends.add(pluginName);
+						softDependencies.put(loadBefore, softDepends);
+					}
+				}
+				catch(InvalidDescriptionException e)
+				{
+					e.printStackTrace();
+				}
+			}
+
+			Set<Plugin> loadedPlugins = new HashSet<>();
+
+			//load in order gathered before
+			while(!pluginsToLoad.isEmpty())
+			{
+				String pluginToLoad = pluginsToLoad.keySet().iterator().next();
+				loadedPlugins.addAll(loadPluginWithDependencies(pluginToLoad, pluginsToLoad, dependencies, softDependencies, 10));
+			}
+			return loadedPlugins.toArray(Plugin[]::new);
+		} 
+		catch (IOException | InvalidPluginException e)
+		{
+			e.printStackTrace();
+		}
+		return new Plugin[0];
+	}
+
+	private void checkPluginNameValidity(File originFile, String pluginName) throws InvalidDescriptionException 
+	{
+		for (String restrictedName : restrictedPluginNames) 
+		{
+			if (pluginName.equalsIgnoreCase(restrictedName)) 
+			{
+				throw new InvalidDescriptionException(String.format("Plugin %s wasn't loaded: The plugin name %s is forbidden!", originFile.getName(), pluginName));
+			}
+		}
+	
+		if (pluginName.contains(" "))
+		{
+			throw new InvalidDescriptionException(String.format("Plugin %s wasn't loaded: The plugin name isn't allowed to have spaces!", originFile.getName()));
+		}
+	}
+
+	private Set<Plugin> loadPluginWithDependencies(String pluginToLoad, Map<String, File> pluginsToLoad, Map<String, Set<String>> dependencies, Map<String, Set<String>> softDependencies, int recursiveDepthLeft) throws InvalidPluginException
+	{
+		Set<Plugin> result = new HashSet<>();
+		result.addAll(loadDependencies(pluginToLoad, pluginsToLoad, dependencies, dependencies, softDependencies, recursiveDepthLeft));
+		try{
+			result.addAll(loadDependencies(pluginToLoad, pluginsToLoad, softDependencies, dependencies, softDependencies, recursiveDepthLeft));
```

Comment:
You're loading all plugin dependencies twice. Did you mean for it to work like this?